node/tools/src/main.rs

```diff
@@ -41,6 +46,19 @@ impl Args {
     }
 }
 
+/// Parse a single (SocketAddr,node key) pair
+fn parse_seed_peer(
+    s: &str,
+) -> Result<(std::net::SocketAddr, node::PublicKey), Box<dyn Error + Send + Sync>> {
+    let pos = s
+        .find(',')
+        .ok_or_else(|| format!("expected format: <IP>:<Port>,<key>"))?;
```

Comment:
nit: consider using json encoding rather than manual parsing for the flag values. You can even reuse the Addr struct from the config.
nit: alternatively please consider whether passing just the whole json encoded config as a flag wouldn't be simpler and more flexible

Reply:
I think it will be good to use json encoding for these pair (SocketAddr, PublicKey) values. It will make the command line argument more consistent and easy to build. 

It may be also useful to pass the whole config as a json encoded, but I don't think it will be necessary for this case and it may overcomplicate the k8s entrypoint call, as well as the file-config vs. cli-arg overriding code.

So if it is ok with you I think I will just implement the json encoding for the `gossip_static_outbound` field.

In order to do this, I'm not sure I understand what Addr struct you are referring to. Is it proto::NodeAddr? Not sure if it is straightforward to deserialize it (maybe I have to make it implement some trait) and even if I do, it will return strings instead of PublicKey and SocketAddr.

Maybe it's cleaner to use a new custom struct for this?

Reply:
Yes, I  meant `proto::NodeAddr`. Currently it is decoded directly in the `config::AppConfig` deserializer: https://github.com/matter-labs/era-consensus/blob/a766bd2b981e08eb106c3c787cca023f7d8f48e4/node/tools/src/config.rs#L65 , but you can create a dedicated `config::NodeAddr` struct and implement `ProtoFmt` trait for it (by just extracting the related code from `ProtoFmt` implementation for `config::AppConfig`). The json decoding function is here: https://github.com/matter-labs/era-consensus/blob/a766bd2b981e08eb106c3c787cca023f7d8f48e4/node/tools/src/config.rs#L18